scene/main/scene_tree.cpp

```diff
@@ -1953,6 +1954,32 @@ bool SceneTree::is_using_font_oversampling() const {
 	return use_font_oversampling;
 }
 
+void SceneTree::get_argument_options(const StringName &p_function, int p_idx, List<String> *r_options) const {
+
+	String pf = p_function;
+	if (p_function == "change_scene") {
+		List<EditorFileSystemDirectory *> directories;
```

Comment:
You can't use editor-specific types in `scene/`, as they don't exist in `tools=no` builds. So you should enclose this code and includes in `#ifdef TOOLS_ENABLED`.

I'm not sure but the signature itself might still need to be exposed to non-tools builds as a stub to ensure that the tools and non-tools core API hashes are matching, CC @neikeq.

Reply:
Ok, I wrapped code of the method in #ifdef. Without tools it does nothing.

Reply:
Actually it turns out that we shouldn't add `editor/` dependencies to `scene/` at all, see #29730. So we need another way to get this.

Reply:
Changed it to regular DirAccess.